src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+/**
+ * 
+ * Deals with making sense of the user's command.
+ * It takes in user inputs and filters it to its respective command. 
+ */
+public class Parser {
+    public Parser() {
+
+    }
+
+    /**
+     * 
+     * 
+     * @param command Command is the user input.
+     * @param taskArr List of tasks in a TaskList. 
+     * @throws DukeException If input is unexpected or invalid.
+     */
+    public Command parse(String command, TaskList taskArr) throws DukeException{
+        String[] commandArr = command.split(" ");
+        if (command.equals("list")) {
+            return new ListCommand();
+        } else if (commandArr[0].equals("done")) {
+            int taskArrRef = Integer.parseInt(commandArr[1]) - 1;
+            return new DoneCommand(taskArrRef);
+        } else if (commandArr[0].equals("delete")) {
+            int taskArrRef = Integer.parseInt(commandArr[1]) - 1;
+            return new DeleteCommand(taskArrRef);
+        } else if (commandArr[0].equals("find")) {
+            try {
+                String keyword = commandArr[1];
+                return new FindCommand(keyword);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Please insert a word after \"done\"");
+            }
+            
+        } else {
+            boolean wrongArrayLength = commandArr.length <= 1;
+            String taskAdded = "Got it. I've added this task: ";
+            if (commandArr[0].equals("todo")) {
+                if (wrongArrayLength) {
+                    throw new DukeException("The description of a todo cannot be empty!");
+                } else {
+                    System.out.println(taskAdded);
+                    int spaceIndex = command.indexOf(" ");
+                    Task task = new Todo(command.substring(spaceIndex + 1));
+                    return new TodoCommand(task);
+                }
+
+            } else if (commandArr[0].equals("deadline")) {
+                if (wrongArrayLength) {
+                    throw new DukeException("The description of a deadline cannot be empty!");
+                } else if (!command.contains("/by ")) {
+                    throw new DukeException("Remember to enter deadline in this format:\"[deadline] [task] /by [date]\"");
+                } else {
+                    System.out.println(taskAdded);
+                    int spaceIndex = command.indexOf(" ");
+                    int slashIndex = command.indexOf("/by ");
+                    Task task = new Deadline(command.substring(spaceIndex + 1, slashIndex - 1), command.substring(slashIndex + 4));
```

Comment:
You can refactor this part by using split
```suggestion
                    String[] args = command.split(" /by ");
                    String description = args[0];
                    String date = args[1];
                    Task task = new Deadline(description, date);
```